src/controllers/job-application.controller.ts

```diff
@@ -1,46 +1,49 @@
 import { Request, Response } from 'express';
-import { JobApplication } from '../models/job-application.model';
+import { applyForJobService, getJobApplicationsByJobIdService } from '../services/job-application.service';
 
 export const applyForJob = async (req: Request, res: Response) => {
 	try {
-		//Extract fields from request body
+		// Extract fields from request body
 		const { job_id, job_seeker_id, cover_letter, resume } = req.body;
 
 		// Basic validation to ensure all required fields are provided
 		if (!job_id || !job_seeker_id || !cover_letter || !resume) {
 			return res.status(400).json({ message: 'All fields are required' });
 		}
 
-		// Log the request body
-		console.log('Request body:', req.body);
-
-		// Check if there is already an application from this job seeker for the specified job
-		console.log('Checking for existing application...');
-		const existingApplication = await JobApplication.findOne({ job_id, job_seeker_id });
-		if (existingApplication) {
-			console.log('Existing application found:', existingApplication);
-			return res.status(400).json({ message: 'Job already applied for' });
-		}
-
-		// Create a new job application entry
-		console.log('Creating new application...');
-		const newApplication = new JobApplication({ job_id, job_seeker_id, cover_letter, resume });
-
-		// Save the new application to the database
-		const savedApplication = await newApplication.save();
-
-		// Log the saved application details for debugging purposes
-		console.log('Saved application:', savedApplication);
+		// Call service to handle the business logic
+		const savedApplication = await applyForJobService(job_id, job_seeker_id, cover_letter, resume);
```

Comment:
The resume is actually a file(pdf, word) and not a string. You should upload a resume to a cloud storage and return a url to the file which should be stored. 
When you pull, go through the code and see how I handled the file upload.